### ISSUE TITLE
Focus stealing bug

### ISSUE THREAD

**Issue by The-Compiler** [herbstluftwm/herbstluftwm#33]
Sometimes, dwb or mpv steal the focus, and switching keyboard focus does not work (immediately focuses back to dwb/mpv again).

Maybe related (?): Some clients (steam) create windows which have focus without having the focus borde

**Comment by fdietze** [comment]
I can reproduce this using entr (http://entrproject.org/) and their reload-browser script with Chromium and Firefox.

**Comment by The-Compiler** [comment]
I just saw the same thing happening with two urxvt instances, so it doesn't seem to be connected to those applications.

**Comment by ypnos** [comment]
I found an easy method to reproduce X/hlwm focus mismatch. Use `xdo-tool` to hand focus to a specific window. Hlwm fails to reflect the change and redraw borders.

**Comment by ypnos** [comment]
I recently experienced another focus problem with a Java application. Lets say I have a layout A | B. In B the java application window resides. In A a terminal window resides.
Now I have the terminal window focused and move it over to B.

While the terminal is drawn as expected, all keyboard input actually goes to the hidden java application. I have to cycle between them to get focus back to the terminal.

**Comment by t-wissmann** [comment]
I think this now has been fixed (at least in the instances that I could reproduce via xdotool and emacsclient). If it still happens with the current git version or with the next release (0.9.4 or higher), please re-open.
